tensorflow/core/kernels/mkl/mkl_matmul_ops_common.h

```diff
@@ -809,11 +809,14 @@ class MklMatMulPrimitive : public MklPrimitive {
     return context_.prim_desc->scratchpad_desc();
   }
   void Execute(const std::shared_ptr<stream>& stream, const Tlhs* a_data,
-               const Trhs* b_data, const Toutput* c_data, void* sp_data,
-               void* mul_data = nullptr, void* add_data = nullptr) {
+               const Trhs* b_data, const Toutput* c_data,
+               const MklMatMulParams& matmul_params, void* sp_data,
+               const std::vector<void*> binary_op_fusions_data = {}) {
```

Comment:
Are you sure you want to pass a vector by value?

Reply:
This is still unresolved.  You shouldn't really be creating empty vectors for default arguments here, and I don't think you want to be passing vectors by value.  I suggest either refactoring so that all calls pass in a vector by const-reference, or if you want to keep an "optional" parameter, writing two overloads.

Reply:
I have changed it to const reference and and removed default arguments. Thanks!